python/tvm/relay/ad.py

```diff
@@ -0,0 +1,12 @@
+# pylint: disable=no-else-return
+# pylint: disable=unidiomatic-typecheck
+"""
+The set of automatic differentiation algorithms in Relay.
+"""
+from . import _ad
+
+def fo_with_gradient(expr, mod=None):
+    return _ad.fo_with_gradient(expr, mod)
+
+def with_gradient(expr, mod=None):
```

Comment:
Personally prefer `gradient` directly. And can we make the `order` as an argument? like `gradient(expr, mod=None, order=1)`, although I am not sure whether the gradient function with different order can share the same interface

Reply:
higher order gradient is blocked until we has reference (which I will pr after type inference).
fo_with_gradient is a method that work only on first order code, and dont use reference, hence the difference.

Reply:
We can just standardize the interface @MarisaKirisame, @ZihengJiang we could use `gradient` name for the operator, the plan was to expose `relay.op.grad(...)` which is expanded to the gradient during a later pass.

Reply:
@MarisaKirisame My point is the interface, even without high order gradient implementation now.